instrumentor/controllers/sources_webhooks.go

```diff
@@ -54,8 +57,17 @@ func (s *SourcesDefaulter) Default(ctx context.Context, obj runtime.Object) erro
 		source.Labels = make(map[string]string)
 	}
 
-	if _, ok := source.Labels[k8sconsts.WorkloadNameLabel]; !ok {
-		source.Labels[k8sconsts.WorkloadNameLabel] = source.Spec.Workload.Name
+	// Set the workload name label - use hash for regex patterns since Kubernetes labels
+	// cannot contain regex special characters like *
+	if source.Spec.UseRegex {
+		// For regex sources, use a hash of the pattern as the label value
+		hash := sha256.Sum256([]byte(source.Spec.Workload.Name))
+		source.Labels[k8sconsts.WorkloadNameLabel] = "regex-" + hex.EncodeToString(hash[:])[:16]
+	} else {
+		// For non-regex sources, use the exact workload name
+		if _, ok := source.Labels[k8sconsts.WorkloadNameLabel]; !ok {
+			source.Labels[k8sconsts.WorkloadNameLabel] = source.Spec.Workload.Name
```

Comment:
the code [here](https://github.com/odigos-io/odigos/blob/e9f645e3d40c695ae6c108b10f495be25867961f/frontend/graph/loaders/fecthers.go#L117) also relies on the existing labels semantics and need to include this new option for matching workload with sources.